tests/test.c

```diff
@@ -24,134 +24,39 @@
-        BUG();
-    }
-
-    printk("Got CPU string '%s' and frequency '%llu'\n", cpu_str, result);
-    return;
-}
-
-static int __user_text func(void *arg) { return 0; }
-
-static int ktf_unit_tests(void) {
-    usermode_call(func, NULL);
-
-    printk("\nLet the UNITTESTs begin\n");
-    printk("Commandline parsing: %s\n", kernel_cmdline);
-
-    if (strcmp(opt_string, "foo")) {
-        printk("String parameter opt_string != foo: %s\n", opt_string);
-        BUG();
-    }
-    else {
-        printk("String parameter parsing works!\n");
-    }
-
-    if (strcmp(opt_badstring, "tool")) {
-        printk("String parameter opt_badstring != tool: %s\n", opt_badstring);
-        BUG();
-    }
-    else {
-        printk("String parameter parsing works!\n");
-    }
-
-    if (opt_ulong != 42) {
-        printk("Integer parameter opt_ulong != 42: %d\n", opt_ulong);
-        BUG();
-    }
-    else {
-        printk("Integer parameter parsing works!\n");
-    }
-
-    if (!opt_bool || !opt_booltwo) {
-        printk("Boolean parameter opt_bool != true: %d\n", opt_bool);
-        printk("Boolean parameter opt_booltwo != true: %d\n", opt_booltwo);
-        BUG();
-    }
-    else {
-        printk("Boolean parameter parsing works!\n");
-    }
-
-    printk("\nMemmove testing:\n");
-    (void) memmove(memmove_string, opt_string, sizeof(opt_string));
-    if (!strcmp(memmove_string, opt_string)) {
-        printk("Moving around memory works!\n");
-    }
-    else {
-        printk("Memmove'ing did not work: %s (%p) != %s (%p)\n", memmove_string,
-               memmove_string, opt_string, opt_string);
-    }
-
-    src = (char *) range_string;
-    dst = (char *) range_string + 2;
-    (void) memmove(dst, src, 4);
-    if (!strcmp(range_string, "121234")) {
-        printk("Moving around memory with overlaping ranges works!\n");
-    }
-    else {
-        printk("Overlaping memmove'ing did not work: %s != %s\n", range_string, "121234");
-    }
-
-    cpu_freq_expect("Intel(R) Xeon(R) CPU E3-1270 V2 @ 3.50GHz", 3500000000);
-    cpu_freq_expect("Intel(R) Celeron(R) CPU J1900 @ 1.99GHz", 1990000000);
-    cpu_freq_expect("AMD G-T48L Processor", 0);
-    cpu_freq_expect("Intel(R) Atom(TM) CPU E3815 @ 1.46GHz", 1460000000);
-    cpu_freq_expect("Intel(R) Atom(TM) CPU E620 @ 600MHz", 600000000);
-    cpu_freq_expect("VIA C7 Processor 1000MHz", 1000000000);
-    cpu_freq_expect("Intel(R) Core(TM) i7 CPU 950 @ 3.07GHz", 3070000000);
-    cpu_freq_expect("AMD Ryzen Threadripper 1950X 16-Core Processor", 0);
-    cpu_freq_expect("Prototyp Amazing Foo One @ 1GHz", 1000000000);
-    cpu_freq_expect("Prototyp Amazing Foo Two @ 1.00GHz", 1000000000);
-
-    printk("Long mode to real mode transition:\n");
-    long_to_real();
+static const char opt_test_delims[] = ",";
+#include <cmdline.h>
 
-    return 0;
-}
-#else
-static int ktf_unit_tests(void) { return 0; }
-#endif
+static char opt_tests[128];
```

Comment:
magic number, please add define and a comment why 128

Reply:
128 is a guess for a length that should cover everything for a while; it should be checked at static time anyway